src/main/java/com/teradata/tpcds/ScalingInfo.java

```diff
@@ -69,21 +69,41 @@ public long getRowCountForScale(int scale)
         }
     }
 
-    private long computeCountUsingStaticScale()
+    private long computeCountUsingStaticScale(float scale)
     {
-        return getRowCountForScale(1);
+        if (scale == Math.round(scale)) {
+            return getRowCountForScale(1);
+        }
+        // Handle decimal scales (not part of spec). Will only have an effect on
+        // decimal scales 0 < x < 1.
```

Comment:
How far off are we?  Because e.g. for the date_dim table there's no randomness in the distribution of the rows, you just go in order and generate  a row for every date from 1/1/1900-1/1/2100.  So if you scale it down for .1 scale, you're making the date range 1/10 of the size (so until 1920), which will have a very substantial effect on queries.